client.py

```diff
@@ -84,30 +84,30 @@ def parse_ServerResponse(self, response):
 if args.verbose:
     print(res1)
 
-session_id = res1.getroot().attrib['SessionID']
+session_id = res1.attrib['SessionID']
 k = res1.find('.//{http://www.w3.org/2000/09/xmldsig#}RSAKeyValue')
 mod = number.bytes_to_long(d64sb(k.find(
   '{http://www.w3.org/2000/09/xmldsig#}Modulus').text))
 exp = number.bytes_to_long(d64sb(k.find(
   '{http://www.w3.org/2000/09/xmldsig#}Exponent').text))
-pubk = RSA.construct({'n': mod, 'e': exp})
+pubk = RSA.construct( (int(mod), int(exp)) )
 pl = res1.find('.//Payload')
 server_nonce = d64sb(pl.find('Nonce').text)
 
 print("Got server nonce and RSA pubkey:\n{}\n{}".format(
     hexlify(server_nonce), pubk.exportKey('PEM').decode()))
 
 # generate and encrypt client nonce
-client_nonce = random.getrandbits(16*8)
+client_nonce = bytearray([random.getrandbits(8) for i in range(16)])
 cipher = PKCS1_OAEP.new(pubk)
-encrypted_client_nonce = cipher.encrypt(client_nonce)
+encrypted_client_nonce = cipher.encrypt(hexlify(client_nonce))
```

Comment:
with or without this I get "Failed to process CT-KIP clientNonceRequest. Status code = MalformedRequest"

hmm